rtrlib/spki/hashtable/ht-spkitable.c

```diff
@@ -73,212 +73,213 @@ static void spki_record_to_key_entry(struct spki_record *spki_r,
+                             struct spki_record **result,
+                             unsigned int *result_size)
 {
-	struct key_entry *current_entry;
-	tommy_node *current_node;
-	void *tmp;
-	*result = NULL;
-	*result_size = 0;
-
-	pthread_rwlock_rdlock(&spki_table->lock);
-
-	current_node = tommy_list_head(&spki_table->list);
-	while (current_node) {
-		current_entry = (struct key_entry *)current_node->data;
-
-		if (memcmp(current_entry->ski, ski,
-			   sizeof(current_entry->ski)) == 0) {
-			(*result_size)++;
-			tmp = realloc(*result,
-				      sizeof(**result) * (*result_size));
-			if (tmp == NULL) {
-				free(*result);
-				pthread_rwlock_unlock(&spki_table->lock);
-				return SPKI_ERROR;
-			}
-			*result = tmp;
-			key_entry_to_spki_record(current_entry,
-						 *result + (*result_size-1));
-		}
-		current_node = current_node->next;
-	}
-	pthread_rwlock_unlock(&spki_table->lock);
-	return SPKI_SUCCESS;
+    struct key_entry *current_entry;
+    tommy_node *current_node;
+    void *tmp;
+    *result = NULL;
+    *result_size = 0;
+
+    pthread_rwlock_rdlock(&spki_table->lock);
+
+    current_node = tommy_list_head(&spki_table->list);
+    while (current_node) {
+        current_entry = (struct key_entry *)current_node->data;
+
+        if (memcmp(current_entry->ski, ski, sizeof(current_entry->ski)) == 0) {
+            (*result_size)++;
+            tmp = realloc(*result,
+                      sizeof(**result) * (*result_size));
+            if (tmp == NULL) {
+                free(*result);
+                pthread_rwlock_unlock(&spki_table->lock);
+                return SPKI_ERROR;
+            }
+            *result = tmp;
+            key_entry_to_spki_record(current_entry, *result+(*result_size-1));
+        }
+        current_node = current_node->next;
+    }
+    pthread_rwlock_unlock(&spki_table->lock);
+    return SPKI_SUCCESS;
 }
 
 int spki_table_remove_entry(struct spki_table *spki_table,
-			    struct spki_record *spki_record)
+                            struct spki_record *spki_record)
 {
-	uint32_t hash;
-	struct key_entry entry;
-	struct key_entry *rmv_elem;
-	int rtval;
-
-	spki_record_to_key_entry(spki_record, &entry);
-	hash = tommy_inthash_u32(spki_record->asn);
-
-	pthread_rwlock_wrlock(&spki_table->lock);
-
-	if (!tommy_hashlin_search(&spki_table->hashtable, spki_table->cmp_fp,
-				  &entry, hash)) {
-		rtval = SPKI_RECORD_NOT_FOUND;
-		goto out;
-	}
-
-	/* Remove from hashtable and list */
-	rmv_elem = tommy_hashlin_remove(&spki_table->hashtable,
-					spki_table->cmp_fp, &entry, hash);
-	if (rmv_elem && tommy_list_remove_existing(&spki_table->list,
-						   &rmv_elem->list_node)) {
-		free(rmv_elem);
-		spki_table_notify_clients(spki_table, spki_record, false);
-		rtval = SPKI_SUCCESS;
-	}
-
-out:
-	pthread_rwlock_unlock(&spki_table->lock);
-	return rtval;
+    uint32_t hash;
+    struct key_entry entry;
+    struct key_entry *rmv_elem;
+    int rtval = SPKI_SUCCESS;
```

Comment:
Variables that are allocated on the stack are never initialized by the compiler.
The value of these variables is unknown!